bindings/azure/cosmosgraphdb/cosmosgraphdb.go

```diff
@@ -0,0 +1,123 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package cosmosgraphdb
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+	gremcos "github.com/supplyon/gremcos"
+)
+
+const (
+	queryOperation bindings.OperationKind = "query"
+
+	// keys from request's Data
+	commandGremlinKey = "gremlin"
+
+	// keys from response's Data
+	respGremlinKey   = "gremlin"
+	respOpKey        = "operation"
+	respStartTimeKey = "start-time"
+	respEndTimeKey   = "end-time"
+	respDurationKey  = "duration"
+)
+
+// CosmosGraphDB allows performing state operations on collections
+type CosmosGraphDB struct {
+	metadata *cosmosGraphDBCredentials
+	client   *gremcos.Cosmos
+	logger   logger.Logger
+}
+
+type cosmosGraphDBCredentials struct {
+	URL       string `json:"url"`
+	MasterKey string `json:"masterKey"`
+	Username  string `json:"username"`
+}
+
+// NewCosmosGraphDB returns a new CosmosGraphDB instance
+func NewCosmosGraphDB(logger logger.Logger) *CosmosGraphDB {
+	return &CosmosGraphDB{logger: logger}
+}
+
+// Init performs CosmosDB connection parsing and connecting
+func (c *CosmosGraphDB) Init(metadata bindings.Metadata) error {
+	c.logger.Debug("Initializing Cosmos Graph DB binding")
+
+	m, err := c.parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	c.metadata = m
+	client, err := gremcos.New(c.metadata.URL,
+		gremcos.WithAuth(c.metadata.Username, c.metadata.MasterKey),
+	)
+	if err != nil {
+		return errors.New("CosmosGraphDB Error: failed to create the Cosmos Graph DB connector")
+	}
+
+	c.client = client
+
+	return nil
+}
+
+func (c *CosmosGraphDB) parseMetadata(metadata bindings.Metadata) (*cosmosGraphDBCredentials, error) {
+	b, err := json.Marshal(metadata.Properties)
+	if err != nil {
+		return nil, err
+	}
+
+	var creds cosmosGraphDBCredentials
+	err = json.Unmarshal(b, &creds)
+	if err != nil {
+		return nil, err
+	}
+
+	return &creds, nil
+}
+
+func (c *CosmosGraphDB) Operations() []bindings.OperationKind {
+	return []bindings.OperationKind{queryOperation}
+}
+
+func (c *CosmosGraphDB) Invoke(req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	var jsonPoint map[string]interface{}
+	err := json.Unmarshal(req.Data, &jsonPoint)
+	if err != nil {
+		return nil, errors.New("CosmosGraphDB Error: Cannot convert request data")
+	}
+
+	gq := fmt.Sprintf("%s", jsonPoint[commandGremlinKey])
```

Comment:
Nit: It might more sense to unmarshal into an inline struct instead of `map[string]interface{}` since you know the field you expect.

```golang
type gremlinData struct {
  Gremlin string `json:"gremlin"` // Note sure if this is always a string, use `interface{}` otherwise.
}
var jsonPoint gremlinData
```

Also, using `%v` might make more since for an `interface{}` going in. But again, if the value is always going to be a `string` then the `fmt.Sprintf` is unnecessary.

```golang
gq := fmt.Sprintf("%v", jsonPoint[commandGremlinKey])
```